app/services/supplementary-billing/new-billing-batch.service.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+/**
+ * Top level service for creating and processing a new billing batch
+ * @module NewBillingBatchService
+ */
+
+const BillingPeriodsService = require('./billing-periods.service.js')
+const CreateBillingBatchPresenter = require('../../presenters/supplementary-billing/create-billing-batch.presenter.js')
+const InitiateBillingBatchService = require('./initiate-billing-batch.service.js')
+const ProcessBillingBatchService = require('./process-billing-batch.service.js')
+
+async function go (regionId, userEmail) {
+  const billingPeriods = BillingPeriodsService.go()
+  const financialYearEndings = _financialYearEndings(billingPeriods)
+
+  const billingBatch = await InitiateBillingBatchService.go(financialYearEndings, regionId, userEmail)
+
+  ProcessBillingBatchService.go(billingBatch, billingPeriods)
+
+  return _response(billingBatch)
+}
+
+function _financialYearEndings (billingPeriods) {
+  return {
+    fromFinancialYearEnding: billingPeriods[billingPeriods.length - 1].endDate.getFullYear(),
+    toFinancialYearEnding: billingPeriods[0].endDate.getFullYear()
```

Comment:
We should be able to use `billingPeriods.at(-1)` to get the last item in a slightly cleaner way. We could also use `billingPeriods.at(0)` for consistency.

```suggestion
    fromFinancialYearEnding: billingPeriods.at(-1).endDate.getFullYear(),
    toFinancialYearEnding: billingPeriods.at(0).endDate.getFullYear()
```